File: Strings/Easy/1684. Count the Number of Consistent Strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {

        int counter = 0;
        HashSet<Character> set = new HashSet<>();

        for(int i = 0; i < allowed.length(); i++){
            set.add(allowed.charAt(i));
        }

        for(int i = 0; i < words.length; i++){
            int flag = 0;
            for(int j = 0; j < words[i].length(); j++){
                if(!set.contains(words[i].charAt(j))){
                    flag = 1;
                    break;
                }
            }
            if(flag == 0){
                counter++;
            }
        }

        return counter;








        // int flag = 0;
        // int counter = 0;
        
        // for(int i = 0; i < words.length; i++){
        //     for(int j = 0; j < words[i].length(); j++){
        //         if(allowed.indexOf(words[i].charAt(j)) != -1){
        //             continue;
        //         }
        //         else{
        //             flag = 1;
        //             break;
        //         }
        //     }
        //     if(flag == 0){
        //         counter++;
        //     }
        //     else{
        //         flag = 0;
        //     }
        // }

        // return counter;
    }
}

File: Strings/Easy/Python/1684. Count the Number of Consistent Strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:

        ans = 0

        for w in words:
            flag = 0
            for c in w:
                if c not in allowed:
                    flag = 1
                    break
            
            if flag == 0:
                ans += 1
        
        return ans



        # freq = [0] * 26
        # ans = 0

        # for i in range(0, len(allowed)):
        #     freq[ord(allowed[i]) - ord("a")] += 1
        
        # for s in words:
        #     flag = 0
        #     for i in range(0, len(s)):
        #         if freq[ord(s[i]) - ord("a")] == 0:
        #             flag = 1
        #             break
            
        #     if flag == 0:
        #         ans += 1
        
        # return ans
        
